File: Relogio/source/Relogio.java
import java.util.Scanner;


class Relogio { //para manter duas classes no mesmo arquivo, nao deve ser declarada public

	public int hora;
	public int minuto;
	public int segundo;
	
	public Relogio(int hora, int minuto, int segundo) {
		setSegundo(segundo);
		setMinuto(minuto);
		setHora(hora);
	}
	
	public int getHora() {
		return hora;
	}

	public void setHora(int hora) {
		this.hora = hora;
	}

	public int getMinuto() {
		return minuto;
	}

	public void setMinuto(int minuto) {
		this.minuto = minuto;
	}

	public int getSegundo() {
		return segundo;
	}

	public void setSegundo(int segundo) {
		this.segundo = segundo;
	}

	public void ajustaRelogio(int incrementoHora, int incrementoMinuto, int incrementoSegundo) {
		int segundoAtual = incrementoSegundo + this.getSegundo();
		if (segundoAtual>59) {
			int minutoAdicional = (int) segundoAtual/60;
			incrementoMinuto+= minutoAdicional;
		}
		segundoAtual = segundoAtual % 60;
		this.setSegundo(segundoAtual);
		
		int minutoAtual = incrementoMinuto + this.getMinuto();
		if (minutoAtual>59) {
			int horaAdicional = (int) minutoAtual/60;
				incrementoHora+= horaAdicional;
		}
		minutoAtual = minutoAtual % 60;
		this.setMinuto(minutoAtual);

		int horaAtual = (incrementoHora + this.getHora()) % 24;
			this.setHora(horaAtual);
			
	}
	
	//public void alteraSegundo(int incrementoSegundo) {
	//}
	//public void alteraHora(int incrementoMinuto) {
	//}
	//public void alteraMinuto(int incrementoHora) {
	//}'
	

	public static void main(String[] args) {

		int hora, minuto, segundo;
		int numeroAjustes, i;
		int incrementoHora = 0, incrementoMinuto = 0, incrementoSegundo = 0;
		String incremento;
		Scanner entrada = new Scanner(System.in);

		//System.out.println("Informe hora: ");
		hora = entrada.nextInt();
		//System.out.println("Informe minuto: ");
		minuto = entrada.nextInt();
		//System.out.println("Informe segundo: ");
		segundo = entrada.nextInt();
		//entrada.close();
		Relogio relogio = new Relogio(hora, minuto, segundo);
		
		//System.out.println("Informe numero de parametros de controle de tempo:");
		numeroAjustes = entrada.nextInt();

		for (i = 0; i < numeroAjustes; i++) {
			incremento = entrada.next();
			switch ( incremento ) {
			case "tic":
				incrementoSegundo++;
				break;
			case "tac":
				incrementoMinuto++;
				break;
			default:
				System.out.println("erro");
				numeroAjustes++;
			}
		}
		relogio.ajustaRelogio(incrementoHora, incrementoMinuto, incrementoSegundo);
		System.out.println(relogio.getHora() + ":" + relogio.getMinuto() + ":" + relogio.getSegundo());
	}
}

File: Relogio/source/Relogio.cpp
#include <iostream>
#include <stdio.h>

using namespace std; //isso me livra de tanta resolucao de escopo

class Relogio {
    private:
    int hora;
    int minuto;
    int segundo;
	
	public:
	Relogio(int hora, int minuto, int segundo); //diferente ter que declarar método assim...
	~Relogio();
	int getHora();
	void setHora(int hora);
	int getMinuto();
	void setMinuto(int minuto);
	int getSegundo();
	void setSegundo(int segundo);
	void ajustar(int incrementoHora, int incrementoMinuto, int incrementoSegundo);
};

	Relogio::Relogio(int hora, int minuto, int segundo) { //resolucao de escopo comecou a ficar mais clara
		setHora(hora);
		setMinuto(minuto);
		setSegundo(segundo);
	}
	
	Relogio::~Relogio() {
	}
	
	int Relogio::getHora() {
		return hora;
	}
	
	void Relogio::setHora(int hora) {
		this->hora = hora; //estive me perguntando quando teria que usar '->', agora descobri
	}
	
	int Relogio::getMinuto() {
		return minuto;
	}
	
	void Relogio::setMinuto(int minuto) {
		this->minuto = minuto;
	}
	
	int Relogio::getSegundo() {
		return segundo;
	}
	
	void Relogio::setSegundo(int segundo) {
		this->segundo = segundo;
	}
	
	void Relogio::ajustar(int incrementoHora, int incrementoMinuto, int incrementoSegundo) {
		int segundoAdicional = getSegundo();
		int segundoAtual = incrementoSegundo + getSegundo();
		if (segundoAtual>59) {
			int minutoAdicional = (int) segundoAtual/60;
			incrementoMinuto+= minutoAdicional;
		}
		segundoAtual = segundoAtual % 60;
		setSegundo(segundoAtual);
		
		int minutoAtual = incrementoMinuto + getMinuto();
		if (minutoAtual>59) {
			int horaAdicional = (int) minutoAtual/60;
				incrementoHora+= horaAdicional;
		}
		minutoAtual = minutoAtual % 60;
		setMinuto(minutoAtual);
		
		int horaAtual = (incrementoHora + getHora()) % 24;
		setHora(horaAtual);
	}

int main()
{
    int hora, minuto, segundo;
    int numeroAjustes, i;
    int incrementoHora = 0, incrementoMinuto = 0, incrementoSegundo = 0;
    string incremento;
    cin >> hora;
    cin >> minuto;
    cin >> segundo;
	Relogio relogio(hora, minuto, segundo);
    cin >> numeroAjustes;
	
	for (i = 0; i < numeroAjustes; i++) {
		cin >> incremento;
		if (incremento == "tic") { //tentei fazer switch, mas descobri que nao se pode usar string para case
			incrementoSegundo++;
		} else if (incremento == "tac") {
			incrementoMinuto++;
		} else {
			cout << "erro" << endl;
			numeroAjustes++;
		}
	}
	
	relogio.ajustar(incrementoHora, incrementoMinuto, incrementoSegundo);
	printf("%02d:%02d:%02d\n", relogio.getHora(), relogio.getMinuto(), relogio.getSegundo());
    return 0;
}
